charts/psmdb-operator/templates/role-binding.yaml

```diff
@@ -3,7 +3,11 @@ kind: ServiceAccount
 metadata:
   name: {{ include "psmdb-operator.fullname" . }}
 ---
+{{- if or .Values.watchNamespace .Values.watchAllNamespaces }}
```

Comment:
Should this be `{{ .Values.watchAllNamespaces }}` instead? Or if we wanted to support both values entries for enabling cluster scope, then it'd be `{{- if or not .Values.watchNamespace .Values.watchAllNamespaces }}` to account for negating an empty string given in `watchNamespace`.

We want a `ClusterRoleBinding` for watching all namespaces. So if a `watchNamespace` value is not empty, then we would want a `RoleBinding` to watch the given ns.

Reply:
We either specify that we want to watch all namespaces - we'd need `ClusterRole` to do that, or we specify a single other namespace to watch for (which is a different one from the one we're deployed to, otherwise what's the point) - and we need `ClusterRole` to get access to another namespace too.

We can use `Role` only when we're dealing with the local namespace - and that's where both `watchAllNamespaces` and `watchNamespace` as falsy.

Reply:
The `watchNamespace` handling can definitely be improved to handle more edge cases, such as when it's not empty but is explicitly set to a value equal to `.Release.namespace` - but I didn't bother.

Feel free to pick this code up and apply fixes as needed. Tossing it and rewriting this feature from scratch is also an option I'd be fine with.

Reply:
Thanks for elaborating. 
I wasn't considering the use-case of deploying the operator to a different namespace than where a single psmdb instance would be deployed. I was making the assumption that if you want RBAC outside of the operator's namespace, then it is more than likely needed cluster-wide. So the way you have it makes sense to me now.

When inspecting the [PXC Operator's helm chart](https://github.com/percona/percona-helm-charts/tree/main/charts/pxc-operator), all three use-cases you outlined are supported (ie. same ns; single, different ns than operator's; cluster-wide) through the `watchNamespace` and `watchAllNamespaces` values. Also, they are using the same if statement structure as what you have, [example here](https://github.com/percona/percona-helm-charts/blob/main/charts/pxc-operator/templates/role-binding.yaml#L11). 

With that said, I'd prefer to have consistency across percona helm charts. So since pxc-operator has a working implementation and is a percona chart, I'd be in favor of adopting the approach there, which is what you have, over what is implemented in https://github.com/percona/percona-helm-charts/pull/122.